iree/compiler/Dialect/Flow/Transforms/Passes.cpp

```diff
@@ -114,10 +121,15 @@ void buildFlowTransformPassPipeline(OpPassManager &passManager) {
   passManager.addNestedPass<FuncOp>(mlir::createCanonicalizerPass());
   passManager.addNestedPass<FuncOp>(createFusionOfTensorOpsPass());
   passManager.addNestedPass<FuncOp>(mlir::createCSEPass());
+  if (clEnableLinalgDetensorize) {
+    // NOTE: must run flow->tensor ops and canonicalization after detensorizing.
```

Comment:
comment is not particularly useful/durable - you don't _have_ to run canonicalization for this any more than you have to for anything else, and that it must run before tensor->flow is implicit in its location prior to tensor->flow and adjacency to other ops producing tensor/linalg ops

big structural comments should go on the pass itself - otherwise they get lost in here and someone trying to build their own pipeline would have to copy from this - at which point, they don't need the comments as they are just copying it :P

Reply:
Ok, removed the comment. This pass generates some `tensor` dialect ops that the rest of IREE's compilation pipeline needs to be converted into `flow` ops, then those ops need canonicalization. I'm fine with that being implicit or called out in the places that host those patterns.